fdb-record-layer-core/src/test/resources/OneTypeWithIndexes.json

```diff
@@ -0,0 +1,60 @@
+{
+  "records": {
+    "dependency": [
+      "record_metadata_options.proto"
+    ],
+    "message_type": [
+      {
+        "name": "RecordTypeUnion",
+        "field": [
+          {
+            "name": "_T1",
+            "number": 1,
+            "type": "TYPE_MESSAGE",
+            "type_name": "T1"
+          }
+        ]
+      },
+      {
+        "name": "T1",
+        "field": [
+          {
+            "name": "ID",
+            "number": 1,
+            "type": "TYPE_INT64"
+          },
+          {
+            "name": "VALUE",
+            "number": 2,
+            "type": "TYPE_INT64"
```

Comment:
What would happen here if we had an index set via a field annotation? Would that get added into the `indexes` list (with a name based on the original type) and then type renaming would just keep the old name? Or would the annotation be persisted and potentially re-evaluated leading to an index with the new type name?